### PR TITLE
fix read_rda deprecated error

### PR DESCRIPTION
**Pull request by boathit**
Replace `read_rda` with `FileIO.load`

### CODE REVIEW

**Comment by nalimilan** [comment]
Thanks! I think you need to add `RData` to the `REQUIRES` file to fix the test failure.

**Comment by boathit** [comment]
@nalimilan Yeah, sure. `RData` is required.

**Comment by nalimilan** [comment]
Could you also stop testing Julia 0.3 in `.travis.yml`? RData.jl doesn't support it. Then it should be good to go.

@johnmyleswhite Could you merge this when ready? Then I can tag a release for you. (BTW, you might consider moving the package to JuliaStats if you don't want to have to maintain it.)

**Comment by randyzwitch** [comment]
I can merge and tag when ready

**Comment by nalimilan** [comment]
Cool. It's ready AFAICT.

File: src/RDatasets.jl [hunk 1]
```diff
@@ -1,6 +1,7 @@
 module RDatasets
     using Reexport
     using Compat
+    import FileIO
```
Comment (ararslan) [inline src/RDatasets.jl]: Why `import` instead of `using`?

File: REQUIRE [hunk 2]
```diff
@@ -2,3 +2,4 @@ julia 0.3
 DataFrames
 Reexport
 Compat
+RData
```
Comment (ararslan) [inline REQUIRE]: RData doesn't support Julia 0.3, so you'll have to bump the Julia requirement in REQUIRE as well.

**Comment by ararslan** [comment]
I'm actually a little confused by this. In DataFrames we've deprecated `read_rda` in favor of `FileIO.load`, but it should still be calling `FileIO.load` with a warning. So while this change is good since `read_rda` is deprecated, I think it also reveals an issue in DataFrames.

**Comment by tlnagy** [comment]
bump

**Comment by ararslan** [comment]
I have a tag pending for DataFrames that should fix the error, allowing things to work again. This PR is still good though, since rather than an error it will be a warning. +1 for merging.

**Comment by tlnagy** [comment]
@ararslan which PR are you referring too?

**Comment by ararslan** [comment]
@tlnagy This one, on RDatasets.jl, that changes `read_rda` to `load`.

**Comment by tlnagy** [comment]
Oh shoot. I meant the tag for DataFrames.

**Comment by ararslan** [comment]
https://github.com/JuliaLang/METADATA.jl/pull/6274?

**Comment by tlnagy** [comment]
Yeah, that includes this fix https://github.com/JuliaStats/DataFrames.jl/pull/1056, awesome. Thanks!

**Comment by randyzwitch** [comment]
Alright, merging since you folks say its ok. Will hold off on tagging until tomorrow, just to give everyone a few hours to really make sure this does what you want.

**Comment by tlnagy** [comment]
Can confirm that Gadfly tests pass locally with this change.

**Comment by tlnagy** [comment]
So tagging a new release of `RDatasets` is blocking for getting Gadfly tests to pass due to missing `RData` requirement. The latest release of `DataFrames` v0.8.2 still crashes. 

```julia
WARNING: read_rda(args...) is deprecated, use FileIO.load(args...) instead.
 in depwarn(::String, ::Symbol) at ./deprecated.jl:64
 in read_rda(::String, ::Vararg{String,N}) at ./deprecated.jl:50
 in dataset(::String, ::String) at /Users/tamasnagy/.julia/v0.5/RDatasets/src/dataset.jl:6
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in evalfile(::String, ::Array{String,1}) at ./loading.jl:442
 in run_tests(::String) at /Users/tamasnagy/.julia/v0.5/Gadfly/test/runtests.jl:142
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
while loading /Users/tamasnagy/.julia/v0.5/Gadfly/test/points.jl, in expression starting on line 4
WARNING: FileIO.NotInstalledError(:RData,"")
 in checked_import(::Symbol) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:12
 in #load#17(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:RData}}) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:81
 in load(::FileIO.File{FileIO.DataFormat{:RData}}) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:73
 in #load#13(::Array{Any,1}, ::Function, ::String) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:45
 in load(::String) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:45
 in read_rda(::String, ::Vararg{String,N}) at ./deprecated.jl:52
 in dataset(::String, ::String) at /Users/tamasnagy/.julia/v0.5/RDatasets/src/dataset.jl:6
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in evalfile(::String, ::Array{String,1}) at ./loading.jl:442
 in run_tests(::String) at /Users/tamasnagy/.julia/v0.5/Gadfly/test/runtests.jl:142
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
Library RData is not installed but can load format: FileIO.File{FileIO.DataFormat{:RData}}("/Users/tamasnagy/.julia/v0.5/RDatasets/src/../data/datasets/iris.rda")
FAILED!
ERROR: LoadError: LoadError: RData is not installed.

 in checked_import(::Symbol) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:12
 in #load#17(::Array{Any,1}, ::Function, ::FileIO.File{FileIO.DataFormat{:RData}}) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:81
 in load(::FileIO.File{FileIO.DataFormat{:RData}}) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:73
 in #load#13(::Array{Any,1}, ::Function, ::String) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:45
 in load(::String) at /Users/tamasnagy/.julia/v0.5/FileIO/src/loadsave.jl:45
 in read_rda(::String, ::Vararg{String,N}) at ./deprecated.jl:52
 in dataset(::String, ::String) at /Users/tamasnagy/.julia/v0.5/RDatasets/src/dataset.jl:6
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in evalfile(::String, ::Array{String,1}) at ./loading.jl:442
 in run_tests(::String) at /Users/tamasnagy/.julia/v0.5/Gadfly/test/runtests.jl:142
 in include_from_node1(::String) at ./loading.jl:426
 in include_from_node1(::String) at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
 in process_options(::Base.JLOptions) at ./client.jl:262
 in _start() at ./client.jl:318
 in _start() at /Applications/Julia-0.5.app/Contents/Resources/julia/lib/julia/sys.dylib:?
while loading /Users/tamasnagy/.julia/v0.5/Gadfly/test/points.jl, in expression starting on line 4
while loading /Users/tamasnagy/.julia/v0.5/Gadfly/test/runtests.jl, in expression starting on line 187
===================================================[ ERROR: Gadfly ]====================================================
```

**Comment by ararslan** [comment]
Sounds like DataFrames needs the RData dependency after all? cc @nalimilan and @andreasnoack

**Comment by andreasnoack** [comment]
`RData` is a dependency of `RDatasets` so why is it not being installed?

**Comment by tlnagy** [comment]
Is it a dependency for the tagged version in METADATA.jl? I think that one is still the old version.

**Comment by tlnagy** [comment]
We use the latest tagged version of RDatasets for Gadfly's tests.

**Comment by ararslan** [comment]
> Is it a dependency for the tagged version in METADATA.jl? I think that one is still the old version.

The latest tagged version here includes the RData requirement, and there's a PR for the tag on METADATA. See https://github.com/JuliaLang/METADATA.jl/pull/6279. So the version exists but is not yet installable.

**Comment by tlnagy** [comment]
Looks like there's a versioning problem: https://travis-ci.org/JuliaLang/METADATA.jl/builds/158144062

**Comment by ararslan** [comment]
Ah yes, it should have been tagged as a minor release rather than a patch. cc @randyzwitch

**Comment by randyzwitch** [comment]
Thanks for the notification, I’ll fix this in the next few minutes.

**Comment by randyzwitch** [comment]
Bumped 

https://github.com/JuliaLang/METADATA.jl/pull/6279/files

**Pull request merged**